kube_controller_manager/tests/test_e2e.py

```diff
@@ -0,0 +1,15 @@
+import pytest
+
+from datadog_checks.base import AgentCheck
+
+
+@pytest.mark.e2e
+def test_e2e(dd_agent_check, aggregator, instance):
+
+    with pytest.raises(Exception):
+        dd_agent_check(instance, rate=True)
+    endpoint_tag = "endpoint:" + instance.get('prometheus_url')
```

Comment:
Oops, almost missed this: `dict.get` will by default return `None` for missing keys. This won't work with `+` and a string. Either use square brackets and crash with an obvious error (missing key) or figure out a way to handle missing keys that involves some default string (empty string is an option, I guess).

Reply:
I just tested and it errors out with a `TypeError` for a `NoneType` object:
```
tests/test_e2e.py:14: in test_e2e
   endpoint_tag = "endpoint:" + instance.get('prometheus_url')
   TypeError: can only concatenate str (not "NoneType") to str
```
which should be self explanatory. I think, if we're not passing correct instances to the tests the it should error out the way it does rather than set a default value.

Also, I believe there is an exception that get raised from the base check as well if the `prometheus_url` is missing:
``` 
error: py.loader: could not configure check 'kube_controller_manager (4.3.0)': could not invoke 'kube_controller_manager' python check constructor. New constructor API returned:
Traceback (most recent call last):
  File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/kube_controller_manager/kube_controller_manager.py", line 108, in __init__
    super(KubeControllerManagerCheck, self).__init__(
  File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/checks/openmetrics/base_check.py", line 130, in __init__
    self.get_scraper_config(instance)
  File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/checks/openmetrics/base_check.py", line 152, in get_scraper_config
    raise CheckException("Unable to find prometheus URL in config file.")
datadog_checks.base.errors.CheckException: Unable to find prometheus URL in config file.
```